aidb/engine/engine.py

```diff
@@ -14,7 +14,12 @@ def execute(self, query: str, **kwargs):
     try:
       parsed_query = Query(query, self._config)
       all_queries = parsed_query.all_queries_in_expressions
+
+      # FIXME: check query validity as a whole
+      # check validity of user defined function query
+      _ = parsed_query.udf_query_validity_check
```

Comment:
What's this FIXME?

Reply:
We have many validity checks for different queries, like for approx select, approx aggegation, nested query, named function. They may have some overlap. I think it's better to put them togother and check the validity first.